src/frontend/src/pages/HomePage/GuestHomePage.tsx

```diff
@@ -6,32 +6,63 @@ import { routes } from '../../utils/routes';
 import { useCurrentUser } from '../../hooks/users.hooks';
 import { useEffect } from 'react';
 import { useHomePageContext } from '../../app/HomePageContext';
+import { useCurrentOrganization } from '../../hooks/organizations.hooks';
+import LoadingIndicator from '../../components/LoadingIndicator';
+import ErrorPage from '../ErrorPage';
+import { useGetImageUrl } from '../../hooks/onboarding.hooks';
 
 const GuestHomePage = () => {
   const user = useCurrentUser();
   const history = useHistory();
+  const { data: organization, isError, error, isLoading } = useCurrentOrganization();
   const { setCurrentHomePage } = useHomePageContext();
 
+  const {
+    data: applyInterestImageUrl,
+    isLoading: applyImageLoading,
+    isError: applyImageError
+  } = useGetImageUrl(organization?.applyInterestImageId ?? null);
+  const {
+    data: exploreGuestImageUrl,
+    isLoading: exploreImageLoading,
+    isError: exploreImageError
+  } = useGetImageUrl(organization?.exploreAsGuestImageId ?? null);
+
   useEffect(() => {
     setCurrentHomePage('guest');
   }, [setCurrentHomePage]);
 
+  if (!organization || isLoading || applyImageLoading || exploreImageLoading) return <LoadingIndicator />;
+  if (isError) return <ErrorPage message={error?.message} />;
+  if (applyImageError) return <ErrorPage message="Error loading apply interest image" />;
+  if (exploreImageError) return <ErrorPage message="Error loading explore as guest image" />;
+  if (!applyInterestImageUrl || !exploreGuestImageUrl) throw new Error('Image URLs are undefined');
```

Comment:
woah were not throwing errors in componenets, we should be returning the loading indivator